[FEATURE#128384] feat(sprinter_service_provider): Migration vers Symfony4

aurelienCastellarnau opened the pull request:
- Mise en place de la DI
 - Réécriture du service pour coller à la disparition d'app et l'arrivée du container
 - Ajout d'un dumping de configuration (prepend)
 - Ajout des configurations pour sprinter et pour le producer rabbitMQ de sprinter
 - Ajout de la TestApp
 - Ajout de la configuration des tests behat
 - Test de l'injection du service dans un kernel de test
 - Test du fonctionnement du service

<img width="1440" alt="Screenshot 2019-06-18 at 11 28 23" src="https://user-images.githubusercontent.com/16307418/59670850-eac26b80-91bc-11e9-8a32-d1558e0ae2d1.png">

aurelienCastellarnau committed:
[FEATURE#128384] feat(sprinter_service_provider): Migration vers Symfony4

 - Mise en place de la DI
 - Réécriture du service pour coller à la disparition d'app et l'arrivée du container
 - Ajout d'un dumping de configuration (prepend)
 - Ajout des configurations pour sprinter et pour le producer rabbitMQ de sprinter

 - Missing tests

aurelienCastellarnau committed:
[FEATURE#128384] feat(tests): configuration

 - Ajout de la TestApp
 - Ajout de la configuration des tests

 - Missing features

gdubost commented on .gitignore:
```diff
@@ -2,3 +2,30 @@ bin/
+###< symfony/framework-bundle ###
+
+###> behat/symfony2-extension ###
+behat.yml
```
Ce serait bien d'avoir la conf behat

gdubost commented on behat.yml.dist:
```diff
@@ -0,0 +1,12 @@
+        Behat\Symfony2Extension:
+            kernel:
+                bootstrap: features/bootstrap/bootstrap.php
+                class: App\Kernel
```
Pas besoin de ce fichier

gdubost commented on composer.json:
```diff
@@ -11,27 +12,55 @@
+        "symfony/framework-bundle": "4.2.*",
+        "php-amqplib/rabbitmq-bundle": "^1.14",
+        "symfony/yaml": "4.2.*",
+        "symfony/console": "4.2.*",
```
Je suis pas sur que tu aies besoin de la console

gdubost commented on phpunit.xml.dist:
```diff
@@ -0,0 +1,27 @@
+            <directory>src</directory>
+        </whitelist>
+    </filter>
+</phpunit>
```
Pas besoin de ce fichier

gdubost commented on public/index.php:
```diff
@@ -0,0 +1,27 @@
+$request = Request::createFromGlobals();
+$response = $kernel->handle($request);
+$response->send();
+$kernel->terminate($request, $response);
```
Comme c'est pas une API pas besoin de public/

gdubost commented on src/DependencyInjection/SprinterExtension.php:
```diff
@@ -0,0 +1,98 @@
+    public function prepend(ContainerBuilder $container)
+    {
+        // récupérer le dossier de configs de la lib
+        $configPath  = __DIR__.'/../../config/packages/';
```
Dans une vraie app je suis pas sur que ca marche, parce que la lib sera dans `vendor/etna/sprinter-provider` et si je ne m'abuse config path vaudra `vendor/etna/sprinter-provider/src/DependencyInjection/../../config/packages` donc `vendor/etna/sprinter-provider/config/packages` et je pense pas que ca va marcher comme prévu

aurelienCastellarnau replied:
hmmm du coup tu ne t'abuses pas mais je pense que c'est exactement ce qu'on veut.
Dans le cas d'un chargement de la lib, on chargera d'abord les conf internes de la lib puis on override avec le reste.
Dans mes tests, lorsque j'inclus la lib dans un dossier lib/ cela fonctionne, je ne vois pas qu'elle serait la différence depuis les vendor/ puisque dans ce cas, lib/ et vendor/ sont au meme niveau.

aurelienCastellarnau replied:
problème réglé, la conf est passée direct en php

gdubost commented on .gitignore:
```diff
@@ -2,3 +2,30 @@ bin/
 tmp/
 vendor/
 composer.lock
```
Faut l'enlever ca

Commit:
Modification du bundle

aurelienCastellarnau committed:
[FEATURE#128384] suppression de superflu, commentaires, norme

aurelienCastellarnau committed:
[FEATURE#128384] feat(rebase on Guillaume): PR rebasée sur les corrections de Guillaumme, ajout des modifications liée a cs-fixer, changement dans les dépendances.

aurelienCastellarnau committed:
[FEATURE#128384] feat(test/service/rabbit): Phase intermédiaire d'intégration de rabbitMQ dans les tests

 - On passe par le kernel de la test app pour récupérer le container
 - Cette méthode est supposée permettre d'inclure des services privés mais ce n'est pas probant en l'état
 - Du coup il y'a un appel au service dans un dummy controller pour éviter que le service privé soit supprimé lors du bootstrap car non utilisé
 - Enfin comme rien n'a fonctionné, le service est passé en privé
 - Les paramètres de RabbitMQ sont définis dans des .env
 - Le mieux obtenu est un lancement de phing ok, mais un crash sur l'instanciation avec une erreur dans StreamIO.php

aurelienCastellarnau committed:
feat(test->inject service): le service est reconnu, les tests ne passent pas.

aurelienCastellarnau committed:
FEAT[128384]: tests

Kumatetsu commented on .drone.yml:
```diff
@@ -0,0 +1,20 @@
+image: etna/drone-php7:php71
+env:
```
C'est pas ici qu'il te manque un truc Behat? quand je regarde dans contract API, il y a Ça:   ```
- RABBITMQ_URL=http://guest:guest@localhost:5672
- RABBITMQ_VHOST=/test-behat
```

Je sais pas si c'est ça qiu peut causer ton problème

aurelienCastellarnau committed:
ajout de l'appel au rabbitMq context dans beha.yml

Kumatetsu commented:
Le drone est pas content sur coverall, c'est surement que le token est pas bon. On va bypass et merge soon

gdubost commented on src/DependencyInjection/SprinterExtension.php:
```diff
@@ -0,0 +1,93 @@
+
+        $bundles = $container->getParameter('kernel.bundles');
+        if (!isset($bundles['OldSoundRabbitMqBundle'])) {
+            throw new \Exception('Error Processing Request', 1);
```
Un message un peu plus précis ?

gdubost commented on src/Services/SprinterService.php:
```diff
@@ -0,0 +1,194 @@
+            'Name'    => 'data.csv',
+            'Content' => $csv_base64,
+        ];
+        $queue_opt    = [
```
On a vraiment besoin de ca ?

aurelienCastellarnau committed:
FEAT[128384]: feat(tests): Boom la question du formatage dans la lib sprinter, ajout d'un test qui fail sur l'encodage du json_encode

aurelienCastellarnau committed:
[FEATURE#128384] feat(tests): Fin des tests, coverall a 100%

 - On retire le test sur la présence du bundle rabbitMQ car il est forcément la (et du coup, provoquer l'erreur est pour le moins compliqué a tester)
 - On ajoute des tests sur l'instanciation d'un kernel de test permettant de tester l'injection de dépendance.

gdubost commented on features/bootstrap/bootstrap.php:
```diff
@@ -0,0 +1,4 @@
+<?php
+
+putenv('APP_ENV='.$_SERVER['APP_ENV'] = $_ENV['APP_ENV'] = 'test');
+require dirname(__DIR__, 2).'/config/bootstrap.php';
```
On en a besoin de ce fichier ?

aurelienCastellarnau replied:
Donc non, pas du tout.... :)

gdubost commented on src/Services/SprinterService.php:
```diff
@@ -0,0 +1,105 @@
+            'template'   => $template,
+            'data'       => $data,
+            'printflag'  => $print_flag,
+        ];
```
Le seul truc qui va manquer c'est d'autres paramêtres optionnels qu'on pourrait passer en plus genre `send_mail` ou d'autres

Kumatetsu commented on features/bootstrap/FeatureContext.php:
```diff
@@ -0,0 +1,80 @@
+    /**
+     * @Given je veux envoyer le template :template avec le student :student
+     */
+    public function jsVeuxEnvoyerLeTemplateAvecLeStudent($template, $student_array) {
```
je* (mais la methode d'après c'est pareil donc je me demande si c'est pas voulu?)

Kumatetsu commented on features/bootstrap/FeatureContext.php:
```diff
@@ -0,0 +1,80 @@
+    /**
+     * @Given je veux envoyer le template :template avec le student :student et un print_flag moisi
+     */
+    public function jsVeuxEnvoyerLeTemplateAvecLeStudentEtUnPrintFlagMoisi($template, $student_array) {
```
On veut peut être appeler notre test autrement? :P

aurelienCastellarnau replied:
nan, c'est pas voulu... c'est un cp/cl qui pue

aurelienCastellarnau committed:
[FEATURE#128384] feat(sprinterService): Ajout d'un tableau associatif libre pour passer des options... libres...

```Elle sont libres mes options, il y en a même qui les ont vu volées```

aurelienCastellarnau committed:
[FEATURE#128384] fix(SprinterService): Les options libres sont mergées avec les params
